### PR TITLE
Fix reorder bug

### PR DESCRIPTION
**Pull request by PIC123**
Fixes bug caused by adding event listeners after reordering tags.

### CODE REVIEW

File: src/public/js/video-tagging/optional-tags.html [hunk 1]
```diff
@@ -113,27 +114,35 @@
             Polymer.dom(this.$.buttonsContainer).appendChild(btn);  
         }
 
-        window.addEventListener("keyup", (e) => {
+        if(videotagging.hotkeysAssigned){
+            window.removeEventListener("keyup", this.assignHotkeys);
```
Comment (wbreza) [inline src/public/js/video-tagging/optional-tags.html]: you could avoid the use of `self = ...` if you do a `.bind(this)` on the event handler.

File: src/public/js/video-tagging/optional-tags.html [hunk 2]
```diff
@@ -92,12 +94,11 @@
...
                     this.tagsState[tag] = "full";
                     self.setSelected(e.currentTarget);
-                    self.fire("ontagschanged", { selected: this.projectSelectedTags(), state: this.tagsState});
                 } else {
                     this.tagsState[tag] = "empty";
                     self.setUnselected(e.currentTarget);
-                    self.fire("ontagschanged", { selected: this.projectSelectedTags(), state: this.tagsState});
-                }         
+                } 
+                self.fire("ontagschanged", { selected: this.projectSelectedTags(), state: this.tagsState});
```
Comment (wbreza) [inline src/public/js/video-tagging/optional-tags.html]: Expected to now always fire vs. previously was only firing event in the else case.

File: src/public/js/video-tagging/optional-tags.html [hunk 3]
```diff
@@ -113,27 +114,35 @@
...
+    }, 
+    assignHotkeys(e){
+        let self = videotagging.optionalTags;
             if (e.keyCode >= 49 && e.keyCode <= 57 && !e.shiftKey) {
+                // debugger;
                 let index = e.keyCode - 49;
-                if (index < this.tags.length) {
-                    let tag = this.tags[index];
-                    let btn = this.getBtnByTag(tag);
+                if (index < videotagging.optionalTags.tags.length) {
```
Comment (wbreza) [inline src/public/js/video-tagging/optional-tags.html]: Why the change to use `optionalTags`.  or was `this` previously referring to `videotagging.optionTags` ?

**Comment by pjlittle** [comment]
Nice to see this working.

File: src/public/js/video-tagging/optional-tags.html (hunk 3 above)
Comment (PIC123) [reply src/public/js/video-tagging/optional-tags.html]: yah, in the context of the event listener callback we wanted the this to refer to optional tags
